### ISSUE TITLE
Web project does not release memory

### ISSUE THREAD

**Issue by skidrow406** [dotnet/runtime#95944]
I have the following code:

` var list = new List<TestModel>();
 for(var i = 0; i < count; i++) list.Add(new TestModel());
 list.Clear();
 list = null;`

Why is Sdk="Microsoft.NET.Sdk.Web" memory not released after the above code is finished running? But on Sdk="Microsoft.NET.Sdk.Worker" the memory is released immediately after (TestModel is a class)

**Comment by huoyaoyuan** [comment]
GC is not running immediately for any object. It has its own heuristic and policy to decide when to collect objects. It behaves well if the objects are *eventually* collected during long running.

Even if objects are collected, GC may choose not to return the memory to OS aggressively. It can reserve it for future uses.

Different SDK templates may provide different policies for GC. You can inspect the values in `GCSettings` and `GC.GetConfigurationVariables()`

In most cases, the heuristic of GC works fine. If there's a case when a huge number of objects are released, you may invoke `GC.Collect` to inform it.

**Comment by skidrow406** [comment]
Every time I run the code the memory increases. I have invoke GC.Collect() but the memory is still not released. Will that affect other programs on my server?

**Comment by huoyaoyuan** [comment]
You can inspect memory of the process with more detailed tools. GC may choose to reserve the memory for future use, instead of returning it to OS.

As long as there's an upper limit in an infinite loop, it should be fine. The upper limit will be dynamically adjusted in actual workload.

**Comment by Suchiman** [comment]
@skidrow406 try using 
```csproj
<PropertyGroup>
  <ServerGarbageCollection>false</ServerGarbageCollection>
</PropertyGroup>
```
in your csproj and see if that makes a difference for you.

**Comment by huoyaoyuan** [comment]
Yes, web project uses server GC by default, which has different policies comparing with workstation GC.

**Comment by janvorli** [comment]
@skidrow406 can this issue be closed or do you still experience the problem?
